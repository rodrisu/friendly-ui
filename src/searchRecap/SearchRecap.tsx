import React, { Fragment } from 'react'
import cc from 'classcat'

import { color } from '../_utils/branding'
import { RideAxis } from '../_utils/rideAxis'
import { SearchIcon } from '../icon/searchIcon'
import { Text, TextTagType } from '../text'
import { UneditableTextField } from '../uneditableTextField'
import { StyledSearchRecap } from './SearchRecap.style'

export type SearchRecapProps = Readonly<{
  className?: string
  from?: string
  to?: string
  info?: string
}>

export const SearchRecap = ({ className, from, to, info }: SearchRecapProps) => (
  <StyledSearchRecap className={cc(['kirk-searchRecap', className])}>
    <UneditableTextField addOn={<SearchIcon />} ellipsis>
      <Fragment>
        <Text
          tag={TextTagType.PARAGRAPH}
          textColor={color.midnightGreen}
          className="kirk-requestRecap-route"
        >
          <RideAxis from={from} to={to} />
        </Text>
        {info && (
          <Text tag={TextTagType.SPAN} className="kirk-requestRecap-info">
            {info}
          </Text>
        )}
      </Fragment>
    </UneditableTextField>
  </StyledSearchRecap>
)
