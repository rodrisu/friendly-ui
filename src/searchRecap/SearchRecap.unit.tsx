import React from 'react'
import renderer from 'react-test-renderer'

import { SearchRecap } from './SearchRecap'

const defaultProps = {
  from: 'Middlesbrough',
  to: 'Stoke-on-Trent, city center',
  info: 'Today 2:30pm, 2 passengers',
  large: false,
}

describe('SearchRecap', () => {
  it('Should display searchRecap component', () => {
    const component = renderer.create(<SearchRecap {...defaultProps} />).toJSON()
    expect(component).toMatchSnapshot()
  })
})
