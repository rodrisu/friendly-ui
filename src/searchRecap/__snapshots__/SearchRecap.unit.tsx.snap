// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchRecap Should display searchRecap component 1`] = `
.c3 {
  fill: #708C91;
}

.c3.kirk-icon-wrapper {
  display: inline-block;
  position: relative;
}

.c3.kirk-icon-shadowed {
  border-radius: 50%;
  box-shadow: 0 0 2px rgba(0,0,0,.3);
}

.c3 .kirk-icon-badge {
  position: absolute;
  bottom: 12px;
  left: 12px;
}

.c5 {
  display: inline-block;
  padding: 0.1em 0 0;
  margin: 0 8px;
  height: 0.9em;
  width: 0.9em;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

.c4 {
  margin: 0;
  font-weight: 400;
}

.c4.kirk-text-button {
  color: #054752;
  font-size: 16px;
  line-height: 20px;
}

.c4.kirk-text-body,
.c4.kirk-text-bodyStrong {
  color: #708C91;
  font-size: 16px;
  line-height: 20px;
}

.c4.kirk-text-caption {
  color: #708C91;
  font-size: 13px;
  line-height: 16px;
}

.c4.kirk-text-title,
.c4.kirk-text-titleStrong {
  color: #054752;
  font-size: 18px;
  line-height: 20px;
}

.c4.kirk-text-display2 {
  color: #054752;
  font-size: 82px;
  line-height: 82px;
  font-weight: 500;
}

.c4.kirk-text-display1 {
  color: #054752;
  font-size: 30px;
  line-height: 1.06;
  font-weight: 500;
}

.c4.kirk-text-subheader,
.c4.kirk-text-subheaderStrong {
  color: #054752;
  font-size: 22px;
  line-height: 24px;
}

.c4.kirk-text-title,
.c4.kirk-text-titleStrong {
  color: #054752;
  font-size: 18px;
  line-height: 20px;
}

.c4.kirk-text-bodyStrong,
.c4.kirk-text-subheaderStrong,
.c4.kirk-text-titleStrong {
  font-weight: 500;
}

.c1 {
  padding-left: 24px !important;
  padding-right: 24px !important;
  margin-left: 0;
  margin-right: 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 54px;
  padding: 0 16px;
  background-color: #EDEDED;
  border-radius: 16px;
  color: #054752;
  font-size: 16px;
  line-height: 54px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2 .kirk-uneditableTextField-label:not(:first-child) {
  margin-left: 16px;
}

.c2 .kirk-uneditableTextField-label--ellipsis {
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.c2 .kirk-uneditableTextField-label--placeholder {
  color: #708C91;
}

.c0 .kirk-uneditableTextField {
  width: 100%;
  padding-top: 8px;
  padding-bottom: 8px;
}

.c0 .kirk-uneditableTextField .kirk-icon {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0 .kirk-uneditableTextField-label {
  width: calc(100% - (16px * 3));
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 .kirk-requestRecap-route {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c0 .kirk-requestRecap-locationSeparator {
  margin: 0 4px;
  height: 20px;
  position: relative;
  top: 1px;
}

.c0 .kirk-requestRecap-location {
  display: inline-block;
  text-overflow: ellipsis;
  max-width: calc(50% - (14px / 2) - 4px);
  overflow: hidden;
  white-space: nowrap;
}

.c0 .kirk-requestRecap-location,
.c0 .kirk-requestRecap-info {
  font-size: 13px;
  line-height: 20px;
}

@media (min-width:800px) {
  .c0 .kirk-requestRecap-location,
  .c0 .kirk-requestRecap-info {
    font-size: 16px;
  }

  .c0 .kirk-requestRecap-locationSeparator {
    top: 0;
  }
}

<div
  className="c0 kirk-searchRecap"
>
  <div
    className="c1 kirk-uneditableTextField"
  >
    <div
      className="c2"
    >
      <svg
        aria-hidden={true}
        className="kirk-icon c3"
        height={24}
        viewBox="0 0 24 24"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        
        <g
          fill="none"
          stroke="#708C91"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeMiterlimit="10"
          strokeWidth="1"
        >
          <line
            x1="22"
            x2="16.4"
            y1="22"
            y2="16.4"
          />
          <circle
            cx="10"
            cy="10"
            r="9"
          />
        </g>
      </svg>
      <div
        className="kirk-uneditableTextField-label kirk-uneditableTextField-label--ellipsis"
      >
        <p
          className="kirk-text kirk-text-body c4 kirk-requestRecap-route"
          style={
            Object {
              "color": "#054752",
            }
          }
        >
          <span
            className=""
          >
            <span>
              Middlesbrough
            </span>
            <svg
              aria-hidden={true}
              className="kirk-icon c3 c5"
              height={24}
              viewBox="0 0 24 24"
              width={24}
              xmlns="http://www.w3.org/2000/svg"
            >
              
              <g
                fill="none"
                stroke="#708C91"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeMiterlimit="10"
                strokeWidth="2"
                transform="rotate(180 12 12)"
              >
                <path
                  d="M9 19l-7-7 7-7"
                />
                <path
                  d="M22 12H2"
                />
              </g>
            </svg>
            <span>
              Stoke-on-Trent, city center
            </span>
          </span>
        </p>
        <span
          className="kirk-text kirk-text-body c4 kirk-requestRecap-info"
        >
          Today 2:30pm, 2 passengers
        </span>
      </div>
    </div>
  </div>
</div>
`;
